include/nlohmann/detail/output/binary_writer.hpp

```diff
@@ -1813,8 +1813,18 @@ class binary_writer
                enable_if_t < std::is_signed<C>::value && std::is_unsigned<char>::value > * = nullptr >
     static CharType to_char_type(std::uint8_t x) noexcept
     {
-        static_assert(sizeof(std::uint8_t) == sizeof(CharType), "size of CharType must be equal to std::uint8_t");
+        // Workaround missing "is_trivially_copyable" &&
+        // "is_trivially_default_constructible" in g++ < 5.0.
+        // Fallback to "is_trivial" check.
+        // See <https://github.com/nlohmann/json/issues/4778>.
+#if __GNUG__ && __GNUC__ < 5
```

Comment:
Feels like there should be some `defined()` checks here.  Also not sure why you are checking `__GNUG__` here since this has to be C++.